enhancements/run-bundle.md

```diff
@@ -0,0 +1,263 @@
+  * If `--index-image` is not provided, default to
+    `quay.io/operator-framework/upstream-registry-builder`
+  * If `--index-image` is provided, use the provided value
+* `[--namespace=]` is an optional flag that specifies the namespace to install
+  the operator. It will default to the `KUBECONFIG` context.
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD).
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamspace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precendence order is:
+    * `OperatorGroup`, if it already exists
+    * `AllNamespaces`, if the CSV supports it
+    * `OwnNamespace`, if the CSV supports it
+    * If the CSV does not support either of these, `--install-mode` flag
+      will be **required**
+
+#### Assumptions
+Like the existing `run` command, we **assume** the following:
+
+ 1. OLM is already installed
+ 1. kube API is available with the permissions to create resources
+ 1. kube server can see and pull the provided images (i.e. bundle and index)
+ 1. If an index image is provided, we assume that it has:
+    * bash, so we can change the entrypoint
+    * dependencies required for `opm registry add` so we can inject
+       the bundle into the DB
+    * the existing registry DB located at `/database/index.db`
+
+#### Example UX Scenarios
+
+Running the following scenarios results in different results. While not an
+exhaustive list, these are some example uses of the `run bundle` command.
+
+* `operator-sdk run bundle <bundle-image>` would install into the default kube namespace,
+creates a cluster-wide `OperatorGroup`.
+* `operator-sdk run bundle <bundle-image> --namespace foo` would install into
+  namespace `foo`, creates a cluster-wide `OperatorGroup`
+* `operator-sdka run bundle <bundle-image> --namespace foo --install-mode
+  SingleNamespace=bar` would install into namespace foo, would determine if
+  OperatorGroup exists and is sufficient, if not it would error
+* `operator-sdk run bundle <bundle-image> --install-mode SingleNamespace=bar`
+  would install into the default kube namespace, does the OperatorGroup logic
+
+#### Simple Example
+
+Let's take the simplest example from above.
+
+```
+operator-sdk run bundle <bundle-image>
+```
+
+What happens when the above is invoked?
+
+ 1. Since `--index-image` was not provided, assign variable `registryImage`
+    equal to `quay.io/operator-framework/upstream-registry-builder`
+ 1. Create pod using `registryImage`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported
+    index image
```

Comment:
We probably need to include the pod logs since other errors are possible (e.g. failing to add the bundle to the registry db).

Reply:
Captured that in the examples. Also captured possibly keeping failed pods around in the alternatives section:

## Alternatives

There is a potential alternative to handling failed pods. Instead of cleaning up
the failed pod and capturing the logs, we could keep the failed pod around to
allow developers to capture the logs and debug the issue.

A flag, --skip-cleanup, could be added to skip cleaning up failed pods. This is
similar to scorecard.